WeatherPortal/weatherapp/views.py

```diff
@@ -1,13 +1,48 @@
 import requests
+import os
+import dotenv
+
 from django.contrib.auth import login
 from django.shortcuts import render, redirect
 from django.urls import reverse
+from pprint import pprint
 
 from .forms import CityForm, CustomUserCreationForm
-from .models import City
+from .models import City, Location
+
+API_KEY = os.getenv('API_KEY')
+
+
+def get_city_location(city_name):
+    try:
+        response_city_coords = requests.get(
+            f'http://api.openweathermap.org/geo/1.0/direct?q={city_name}&limit=9&appid={API_KEY}'
+        )
+        get_coords_data = response_city_coords.json()
+        if get_coords_data:
+            return get_coords_data
+        else:
+            return None
+    except requests.exceptions.HTTPError:
+        return "Can't connect with openweather api"
```

Comment:
To nie jest dobry sposób, żeby przekazywać komunikat błędu, możesz wywołać tutaj konkretny błąd z konkretnym messagem i to potem łapać wyżej, zwracanie stringa nie jest ok

Reply:
```
    except requests.exceptions.HTTPError:
        return HttpResponse("Can't connect with openweather api", status=404)
```